src/main/java/org/jenkinsci/maven/plugins/hpi/MavenArtifact.java

```diff
@@ -57,7 +57,12 @@ public MavenArtifact(
     }
 
     public MavenProject resolvePom() throws ProjectBuildingException {
-        return builder.buildFromRepository(artifact,remoteRepositories,localRepository);
+        ProjectBuildingRequest buildingRequest =
+                new DefaultProjectBuildingRequest(session.getProjectBuildingRequest());
+        buildingRequest.setRemoteRepositories(remoteRepositories);
+        buildingRequest.setLocalRepository(localRepository);
```

Comment:
Based on stuff I tried in #66 I suspect these lines could be deleted, and the corresponding fields in `MavenArtifact` deleted; you would still need to duplicate `session.projectBuildingRequest` as you are doing here but only to override `processPlugins`.

Just FYI as a possible cleanup; this PR is directly refactoring the affected code to use the new API which seems fine.

Reply:
I am working on a PR for it.